library/HTMLPurifier/Lexer.php

```diff
@@ -306,8 +306,8 @@ public function normalize($html, $config, $context)
     {
         // normalize newlines to \n
         if ($config->get('Core.NormalizeNewlines')) {
-            $html = str_replace("\r\n", "\n", $html);
-            $html = str_replace("\r", "\n", $html);
+            $html = str_replace("\r\n", "\n", (string)$html);
```

Comment:
Feels like type confusion again. Shouldn't have passed in a non-string html to this function in the first place.

Reply:
I second that - can't find anything within the source code that passes a non-string value (unless user passes a non-string to tokenizeHTML())